peps/pep-0817.rst

```diff
@@ -0,0 +1,2528 @@
+      mission.
+
+    * Undefined behavior due to presence of multiple OpenMP runtimes.
+      XGBoost is installed in a variety of systems with different OpenMP
+      runtimes (or none at all). So far, XGBoost has been vendoring a
+      copy of OpenMP runtime, but this is increasingly untenable. Users
+      get undefined behavior such as crashes or hangs when multiple
+      incompatible versions of OpenMP runtimes are present in the
+      system. (This problem was particularly bad on MacOS, so much so
+      that the MacOS wheel for XGBoost no longer bundles OpenMP.)
+
+    — Philip Hyunsu Cho, a lead maintainer of XGBoost_
+
+The potential for improvement can be summarized as:
+
+    This PEP is a significant step forward in improving the deployment
+    challenges of the Python ecosystem in the face of increasingly
+    complex and varied hardware configurations. By enabling multiple
+    deployment targets for the same libraries in a standard way, it will
+    consolidate and simplify many awkward and time-consuming
+    work-arounds developers have been pursuing to support the rapidly
+    growing AI/ML and scientific computing worlds.
+
+    — Travis Oliphant, the author of NumPy_ and SciPy_ and Chief AI
+    Architect at OpenTeams
+
+
+Motivation summary
+------------------
+
+As highlighted in the previous section, the current Python packaging
+system cannot adequately serve the needs of modern heterogeneous
+computing environments. These aforementioned limitations force package
+authors into complex workarounds that create friction for users,
+increase maintenance burden, and fragment the ecosystem.
+
+**Wheel Variants provide a standardized solution that:**
+
+- Enables automatic hardware-appropriate package selection.
+- Maintains full backward compatibility with existing tools (i.e.,
+  non-variant aware installers, tools, and indexes remain unaffected).
+- Simplifies package maintenance by offering a unified and flexible
+  solution to the challenge of managing multiple platform-specific
+  package builds and distributions.
+- Provides a seamless and predictable experience for users, with
+  little-to-no user input required.
+- Supports the full spectrum of modern computing hardware.
+- Provides a future-proof and flexible system that can evolve with the
+  ecosystem and future use cases.
+
+
+Out-of-scope features
+---------------------
+
+This PEP tries to present the minimal scope required and leaves aspects
+to tools to evolve. A non-exhaustive list:
+
+- The format of a static file to select variants deterministically or
+  include variants in a ``pylock.toml`` file,
+- The list of variant providers that are vendored or re-implemented by
+  installers,
+- The specific opt-in mechanisms and UX for allowing an installer to run
+  non-vendored variant providers,
+- How to instruct build backends to emit variants through the :pep:`517`
+  mechanism.
+
+
+Prior art
+---------
```

Comment:
Move this entire section to an appendix or delete, I don't think it serves to significantly develop the text of the PEP

Reply:
I consider learning from other ecosystems, their experiences and their mistakes a crucial part of any new design. It's important to consider if can adopt an existing, working solution from a different ecosystem with the same problems. The links in these sections were used as references when developing our own design. I'm not particular on placement, but this is not something we can delete.